Feature/modal

atomboulian opened the pull request.

atomboulian commented on package.json:
```diff
@@ -64,11 +64,12 @@
     "react-router-dom": "4.1.2",
     "react-router-redux": "4.0.8",
     "react-scripts": "1.1.0",
-    "react-table": "^6.7.6",
```
Wasn't intending this btw...

mtomcal commented on src/components/layout/Modal.js:
```diff
@@ -0,0 +1,105 @@
+        this.openPortal = this.openPortal.bind(this)
+        this.closePortal = this.closePortal.bind(this)
+        this.wrapWithPortal = this.wrapWithPortal.bind(this)
+        this.handleKeydown = this.handleKeydown.bind(this)
```
You may remove binds now with babel-transform-properties preset I added, if you define your methods like so:

```
handleKeydown = event => {}
```
See also https://github.com/Attainia/attainia-web-components/blob/development/src/components/react-table/TableFieldWithState.js#L23-L29

atomboulian replied:
Understood! Much cleaner now, thank you.

mtomcal commented on src/components/layout/ModalButton.js:
```diff
@@ -0,0 +1,20 @@
+            Open Modal
+            </Button>,
+            portal(<ModalChild closePortal={closePortal} />)
+        ]}
```
Might want to consider making two renderProps. One for renderModalContent and one for renderModalTrigger or some such verbiage. It's an awkward API accepting an Array for this.props.children.

atomboulian replied:
I wanted to use renderProps but I was having trouble wrapping my mind around it when I was reading it yesterday... maybe you can help me make some sense out of it.

Yea, the array for children comes from a different lib I was working off, we totally don't need to keep it.

arizonatribe commented on public/index.html:
```diff
@@ -16,7 +16,8 @@
       You need to enable JavaScript to run this app.
     </noscript>
-    <div id="root"></div>
+    <div id="app-root"></div>
```
If you change this id here, I think it will need to be changed in the `src/globals.js` for this and any other similar component library of ours.

atomboulian replied:
Fixed.

arizonatribe commented on src/App.js:
```diff
@@ -1,13 +1,18 @@
-            <h1>Welcome to our site!</h1>
-        </ContentCentered>
-    </ContentWrapper>
+    <div>
```
Was there a styling issue that caused a wrapper to be needed? Or was it because of the `src/globals.js` still referencing `#root` instead of `#app-root`?

atomboulian replied:
Ah yes, that's probably why!

arizonatribe commented on src/components/layout/Modal.js:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    handleKeydown(e) {
+        if (e.keyCode === KEYCODES.ESCAPE && this.state.active) {
```
Nice

arizonatribe commented on src/components/layout/Modal.js:
```diff
@@ -0,0 +1,105 @@
+  
+    componentWillUnmount() {
+        if (this.props.closeOnEsc) {
+            document.removeEventListener('keydown', this.handleKeydown)
```
Nice! Always good to remember this part

atomboulian replied:
Indeed!

arizonatribe commented on src/components/layout/Modal.js:
```diff
@@ -0,0 +1,105 @@
+    onOpen: PropTypes.func,
+    onClose: PropTypes.func,
+    // node can be a div in index.html instead of a React Node.
+    node: PropTypes.any // eslint-disable-line react/forbid-prop-types
```
Or you can do:

```
  node: PropTypes.node
```

atomboulian replied:
It errors if I use PropTypes.node because the div in question is not a React Node.

atomboulian commented:
I addressed all the feedback besides the render props suggestion. I think this can be our first iteration, and we can definitely do it differently soon.
